Terraform crashes when executing 0.12 upgrade

### Terraform Version

```
Terraform v0.12.0-dev
```

### Terraform Configuration Files

```hcl
provider "aws" {
  region    = "us-west-2"
  profile   = "########"
}

resource "aws_key_pair" "mongodb" {
  key_name   = "#########"
  public_key = "${file("./keys/aws_deployer.pub")}"
}

resource "aws_security_group" "mongodb" {
  name        = "mongodb"
  description = "Allow all inbound SSH and MongoDB traffic."

  ingress {
    from_port   = 22
    to_port     = 22
    protocol    = "tcp"
    cidr_blocks = ["0.0.0.0/0"]
    description = "SSH access"
  }

  ingress {
    from_port   = 27017
    to_port     = 27017
    protocol    = "tcp"
    cidr_blocks = ["0.0.0.0/0"]
    description = "MongoDB access"
  }

  egress {
    from_port       = 0
    to_port         = 0
    protocol        = "-1"
    cidr_blocks     = ["0.0.0.0/0"]
  }

  tags {
    Name = "MongoDB"
  }
}

module "mongo" {
  source = "./modules/mongo"

  aws_region        = "us-east-2"
  availability_zone = "us-east-2a"
  instance_type     = "t2.micro"
  volume_size       = "10"
  key_name          = "${aws_key_pair.mongodb.key_name}"
  private_key       = "${file("./keys/aws_deployer")}"
  security_groups   = ["${aws_security_group.mongodb.name}"]
}
```

### Debug Output

```hcl
-----------------------------------------------------------------------------
2019/04/02 17:17:53 [DEBUG] checking for provider in "."
2019/04/02 17:17:53 [DEBUG] checking for provider in "/Users/justin/go/bin"
2019/04/02 17:17:53 [DEBUG] checking for provisioner in "."
2019/04/02 17:17:53 [DEBUG] checking for provisioner in "/Users/justin/go/bin"
2019/04/02 17:17:53 [TRACE] configupgrade: Analyzing "main.tf"
2019/04/02 17:17:53 [TRACE] Provider block requires provider "aws"
2019/04/02 17:17:53 [TRACE] Resource block for aws_key_pair.mongodb requires provider "aws"
2019/04/02 17:17:53 [TRACE] Resource block for aws_security_group.mongodb requires provider "aws"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x20e3358]

goroutine 1 [running]:
github.com/hashicorp/terraform/configs/configupgrade.(*Upgrader).Upgrade(0xc000557710, 0xc000291740, 0x26ce0cd, 0x1, 0xc0002d8044, 0x3, 0x0, 0x0)
	/Users/justin/go/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade.go:38 +0x78
github.com/hashicorp/terraform/command.(*ZeroTwelveUpgradeCommand).Run(0xc000090d80, 0xc00000c090, 0x0, 0x0, 0xc0002c4260)
	/Users/justin/go/src/github.com/hashicorp/terraform/command/012_config_upgrade.go:153 +0x843
github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc0004bc280, 0xc0004bc280, 0xc00042dd20, 0x1)
	/Users/justin/go/src/github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli/cli.go:255 +0x207
main.wrappedMain(0x0)
	/Users/justin/go/src/github.com/hashicorp/terraform/main.go:223 +0xb37
main.realMain(0x0)
	/Users/justin/go/src/github.com/hashicorp/terraform/main.go:100 +0xb7
main.main()
	/Users/justin/go/src/github.com/hashicorp/terraform/main.go:36 +0x2e
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x20e3358]

goroutine 1 [running]:
github.com/hashicorp/terraform/configs/configupgrade.(*Upgrader).Upgrade(0xc000557710, 0xc000291740, 0x26ce0cd, 0x1, 0xc0002d8044, 0x3, 0x0, 0x0)
	/Users/justin/go/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade.go:38 +0x78
github.com/hashicorp/terraform/command.(*ZeroTwelveUpgradeCommand).Run(0xc000090d80, 0xc00000c090, 0x0, 0x0, 0xc0002c4260)
	/Users/justin/go/src/github.com/hashicorp/terraform/command/012_config_upgrade.go:153 +0x843
github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc0004bc280, 0xc0004bc280, 0xc00042dd20, 0x1)
	/Users/justin/go/src/github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli/cli.go:255 +0x207
main.wrappedMain(0x0)
	/Users/justin/go/src/github.com/hashicorp/terraform/main.go:223 +0xb37
main.realMain(0x0)
	/Users/justin/go/src/github.com/hashicorp/terraform/main.go:100 +0xb7
main.main()
	/Users/justin/go/src/github.com/hashicorp/terraform/main.go:36 +0x2e
```

### Crash Output

```hcl
2019/04/02 17:17:51 [INFO] Terraform version: 0.12.0 dev                                                                                                                      
2019/04/02 17:17:51 [INFO] Go runtime version: go1.11.5
2019/04/02 17:17:51 [INFO] CLI args: []string{"/Users/justin/go/bin/terraform", "0.12upgrade"}
2019/04/02 17:17:51 [DEBUG] Attempting to open CLI config file: /Users/justin/.terraformrc
2019/04/02 17:17:51 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2019/04/02 17:17:51 [INFO] CLI command args: []string{"0.12upgrade"}
2019/04/02 17:17:51 [DEBUG] command: asking for input: "Would you like to upgrade the module in the current directory?"
2019/04/02 17:17:53 [DEBUG] checking for provider in "."
2019/04/02 17:17:53 [DEBUG] checking for provider in "/Users/justin/go/bin"
2019/04/02 17:17:53 [DEBUG] checking for provisioner in "."
2019/04/02 17:17:53 [DEBUG] checking for provisioner in "/Users/justin/go/bin"
2019/04/02 17:17:53 [TRACE] configupgrade: Analyzing "main.tf"
2019/04/02 17:17:53 [TRACE] Provider block requires provider "aws"
2019/04/02 17:17:53 [TRACE] Resource block for aws_key_pair.mongodb requires provider "aws"
2019/04/02 17:17:53 [TRACE] Resource block for aws_security_group.mongodb requires provider "aws"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x20e3358]

goroutine 1 [running]:
github.com/hashicorp/terraform/configs/configupgrade.(*Upgrader).Upgrade(0xc000557710, 0xc000291740, 0x26ce0cd, 0x1, 0xc0002d8044, 0x3, 0x0, 0x0)
  /Users/justin/go/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade.go:38 +0x78
github.com/hashicorp/terraform/command.(*ZeroTwelveUpgradeCommand).Run(0xc000090d80, 0xc00000c090, 0x0, 0x0, 0xc0002c4260)
  /Users/justin/go/src/github.com/hashicorp/terraform/command/012_config_upgrade.go:153 +0x843
github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc0004bc280, 0xc0004bc280, 0xc00042dd20, 0x1)
  /Users/justin/go/src/github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli/cli.go:255 +0x207
main.wrappedMain(0x0)
  /Users/justin/go/src/github.com/hashicorp/terraform/main.go:223 +0xb37
main.realMain(0x0)
  /Users/justin/go/src/github.com/hashicorp/terraform/main.go:100 +0xb7
main.main()
  /Users/justin/go/src/github.com/hashicorp/terraform/main.go:36 +0x2e
```

### Expected Behavior

Successful upgrade to `0.12` syntax in the current module

### Actual Behavior

Terraform crashes

### Steps to Reproduce

`terraform 0.12upgrade`

--- a/configs/configupgrade/upgrade.go
+++ b/configs/configupgrade/upgrade.go
@@ -35,11 +35,11 @@ func (u *Upgrader) Upgrade(input ModuleSources, dir string) (ModuleSources, tfdi
 	var diags tfdiags.Diagnostics
 
 	an, err := u.analyze(input)
-	an.ModuleDir = dir
 	if err != nil {
 		diags = diags.Append(err)
 		return ret, diags
 	}
+	an.ModuleDir = dir
 
 	for name, src := range input {
 		ext := fileExt(name)

